### DISCUSSION TITLE
Anyone have good examples of custom editor components?

### DISCUSSION THREAD

**Discussion by chumbalayaa** [outline/rich-markdown-editor#391] [Q&A]
Using this awesome library in my free time to take my own notes. I'm trying to see what I can do to customize the experience. Some blocks I have thought about:

- Node that is mapped invisibly to an href in my React app, when I hover over the Node a button appears that I can click and navigate to (sort of replicates Notion's sub-page concept)
- Carousel of images rather than being limited to individual images. Perhaps there's some invisible id associated with the images that, when read from the markdown, indicate that the editor should render those images in a carousel
- Live feed of tweets given a specific hashtag or @text input

Anyway, just ideas, but does anyone have examples of their own so I can start from somewhere? Thanks

**Comment by tommoor** [comment]
- I believe the first idea could be achieved with the `onHoverLink` callback, it's designed for pretty much this purpose.
- The carousel is trickier, I can't off the top of my head think of a way to achieve this that isn't structurally a very large change – however I would like to enable arrow <- -> buttons within the image lightbox to scroll through the images in a doc, this would go some way towards achieving it.
- Live feed of tweets would be best to use an embed I'd say :)
